File: kotlin_server/src/main/kotlin/com/jolimoi_test/kotlin_server/service_impl/services/RomanNumeralConvertServiceImpl.kt
package com.jolimoi_test.kotlin_server.service_impl.services

import com.jolimoi_test.kotlin_server.repository.NotifierRepository
import com.jolimoi_test.kotlin_server.repository.NumeralConvertRepository
import com.jolimoi_test.kotlin_server.service_impl.use_cases.convertFromIntegerToRomanNumeralUseCase
import org.springframework.stereotype.Service

@Service(value = "romanNumeralConvertServiceImpl")
class RomanNumeralConvertServiceImpl(private val notifier: NotifierRepository) : NumeralConvertRepository {
    override suspend fun convert(input: Int, clientId: String?): String =
        convertFromIntegerToRomanNumeralUseCase(number = input).let {
            if (clientId != null) {
                notifier.notify(clientId = clientId, data = it)
            }
            it
        }
}

File: node_server/src/api/service_impl/implementations/roman_numeral_convert_service_impl.ts
import NumeralConvertRepository from "@api/repository/numeral_convert_repository";
import convertFromIntegerToRomanNumeralUseCase from "../use_cases/convert_from_number_to_roman_numeral_use_case";

class RomanNumeralConvertServiceImpl extends NumeralConvertRepository {
    convert = (input: number): string => convertFromIntegerToRomanNumeralUseCase(input);
}

export {
    RomanNumeralConvertServiceImpl
}